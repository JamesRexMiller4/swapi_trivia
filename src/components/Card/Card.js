import React from 'react';
import './Card.scss';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Card = (props) => {
// For character cards the props being spread out and passed down will need to be grabbed
// as {id, name, homeworld, species}

  // homeworld, population of homeworld is a url for another fetch request
  // species is another fetch request
  // films is another fetch request
    let card;
    props.type === 'movie' ? card =
      <div className="wrapper-card-div">
        <ul className="card-details-ul">
          <li><strong className="episode-title">Episode {props.id}:</strong> {props.title}</li>
          <li><strong>Release Year:</strong> {props.date}</li>
        </ul>
        <Link to={`/movies/${props.id}`} key={props.id}>
          <button className="character-btn">View Characters</button>
        </Link>
      </div> :
    card =
      <div className="wrapper-card-div">
        <ul className="card-details-ul">
          <li><strong>Name:</strong> Luke SkyWalker</li>
          <li><strong>Homeworld:</strong> Tattoiine</li>
          <li><strong>Population of Tatootine:</strong> 400 million</li>
          <li><strong>Species:</strong> Human</li>
          <li><strong>Other Films:</strong> all seven</li>
        </ul>
      </div>

    return (
      <section className="section-card">
      {card}
      </section>
    )
  }
// }

// For character cards
// return (
//   <section className="section-card">
//   <div className="wrapper-card-div">
//     <ul className="card-details-ul">
//       <li><strong>Name:</strong> Luke SkyWalker</li>
//       <li><strong>Homeworld:</strong> Tattoiine</li>
//       <li><strong>Population of Tatootine:</strong> 400 million</li>
//       <li><strong>Species:</strong> Human</li>
//       <li><strong>Other Films:</strong> all seven</li>
//     </ul>
//   </div>
// </section>
// )

Card.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string,
  date: PropTypes.string,
  characters: PropTypes.array
}



export default Card
