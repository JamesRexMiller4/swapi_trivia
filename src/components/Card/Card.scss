.section-card {
  width: 500px;
  height: 200px;
  background-color: white;
  margin: 20px;
  border: 10px solid;
  color: purple;
}

// @mixin imagegrid($qty, $margin) {
//   width: ((100% - (($qty - 1) * $margin))/ $qty);
//   &:nth-child(n) {
//     margin-right: $margin;
//     margin-bottom: $margin;
//   }
//   &:nth-child(#{qty}n) {
//     margin-right: 0;
//     margin-bottom: 0;
//   }
// }