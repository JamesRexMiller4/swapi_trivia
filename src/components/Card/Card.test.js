import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import {shallow} from 'enzyme';

describe('Card', () => {

  describe('Movie Card', () => {
    let wrapper, charactersArr;

    beforeEach(() => {
      charactersArr = [
        "https://swapi.co/api/people/1/",
        "https://swapi.co/api/people/2/",
        "https://swapi.co/api/people/3/",
        "https://swapi.co/api/people/4/"
      ]

      wrapper = shallow(<Card 
        id={4}
        title={'A NEW HOPE'}
        date={'1977'}
        characters={charactersArr}
        type="movie"
        />)
    });

    it('should match the snapshot with all the data passed through', () => {
      expect(wrapper).toMatchSnapshot()
  });

  describe('Character Card', () => {
    
  })

  });
});