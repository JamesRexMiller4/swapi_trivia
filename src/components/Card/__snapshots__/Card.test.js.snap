// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Movie Card should match the snapshot with all the data passed through 1`] = `
<section
  className="section-card"
>
  <div
    className="wrapper-card-div"
  >
    <ul
      className="card-details-ul"
    >
      <li>
        <strong
          className="episode-title"
        >
          Episode 
          4
          :
        </strong>
         
        A NEW HOPE
      </li>
      <li>
        <strong>
          Release Year:
        </strong>
         
        1977
      </li>
    </ul>
    <Link
      key="4"
      to="/movies/4"
    >
      <button
        className="character-btn"
        id={4}
      >
        View Characters
      </button>
    </Link>
  </div>
</section>
`;
